app/models/manageiq/providers/openstack/cloud_manager/refresh_parser.rb

```diff
@@ -318,12 +318,12 @@ def parse_server(server, parent_hosts = nil)
         :connection_state    => "connected",
 
         :hardware            => {
-          :cpu_sockets          => flavor[:cpus],
+          :cpu_sockets          => flavor.nil? ? nil : flavor[:cpus],
```

Comment:
This can be done cleaner with

```ruby
:cpu_sockets          => flavor && flavor[:cpus],
```

or perhaps even cleaner with

```ruby
:cpu_sockets          => flavor.try([], :cpus),
```